set_boost_charge.py

```diff
@@ -96,28 +97,29 @@ def main():
     server_conn.refresh()
 
     for zappi in server_conn.state.zappi_list():
-        print('Zappi is currently in mode {}'.format(zappi.mode))
-
-        SIDS=[11,12,13,14]
-        if not args.reset:
-            if not zappi.car_connected():
-                print('Setting boost times without car connected?')
-
-            for slot in slots.ranges:
-                duration = slot.duration()
-                server_conn.set_boost(zappi.sno, SIDS.pop(),
-                                      bsh=slot.start_time.tm_hour,
-                                      bsm=slot.start_time.tm_min,
-                                      dow=slot.start_time.tm_wday,
-                                      bdh=duration // 60,
-                                      bdm=duration % 60)
-            price = slots.get_price()
-            total_price = charge_rate / 1000 * price
-            print('Estimated charge cost {:.1f} pence'.format(total_price))
-
-        # Now clear any other boost timers.
-        for zslot in SIDS:
-            server_conn.set_boost(zappi.sno, zslot) 
+        if args.sno == 0 or args.sno == zappi.sno:
+            print('Zappi is currently in mode {}'.format(zappi.mode))
```

Comment:
I would probably have inverted the truth of this if statement, and used 'continue' for other Zappis, then we could avoid extra indentation for the rest of this code, however that's cosmetic.

```suggestion
        if args.sno != and or args.sno != zappi.sno:
            continue
```